src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -159,22 +159,11 @@ async Task EnsureLong()
                     ReadPosition = 0;
                 }
 
-                CancellationTokenSource? combinedCts = null;
-
                 try
                 {
                     var finalCt = cancellationToken;
                     if (async && Timeout > TimeSpan.Zero)
-                    {
-                        _timeoutCts.Start();
-                        finalCt = _timeoutCts.Token;
-
-                        if (cancellationToken.CanBeCanceled)
-                        {
-                            combinedCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _timeoutCts.Token);
-                            finalCt = combinedCts.Token;
-                        }
-                    }
+                        finalCt = _timeoutCts.Start(cancellationToken);
```

Comment:
Unrelated nit, but I think these three lines can move out of the try block (this way, if Start throws we don't end up calling Stop). This is very theoretical and more about general code quality than anything real (it's already this way on the write side).